### ISSUE TITLE
TypeInfo/.init emission and separate compilation

### ISSUE THREAD

**Issue by ariovistus** [ldc-developers/ldc#216]
In the following example, I try to use a module with a struct in it, but not build an object file for the module. This would be ideal for deimos headers. However, link complains. dmd compiles the example just fine. Linux 64, dmd and ldc hot off github.

Is this a thing that could be improved?

test.d:

``` d
import core.vararg;
import cc;

void main () {
    A a;
    void boo(...){
    }
    boo(a);
}
```

The vararg is what triggers the link error.

cc.d:

``` d
extern(C):
    struct A {
        int i;
        int j;
        int k;
    }
```

dmd test
(works)

ldc2 test.d

test.o:(.rodata._D15TypeInfo_S2cc1A6__initZ[_D15TypeInfo_S2cc1A6__initZ]+0x28): undefined reference to `_D2cc1A6__initZ'
collect2: ld returned 1 exit status
Error: /usr/bin/gcc failed with status: 1

**Comment by kinke** [comment]
This hasn't been fixed yet, but needs a non-zeros-only `cc.A.init` to be triggered.

**Comment by kinke** [comment]
If using a `float` field in `cc.A`, there are more linking issues (`xtoHash`, `xopEquals`).

**Comment by kinke** [comment]
This is fixed, just verified with LDC 1.5. The `TypeInfo_Struct` is emitted into the `test` object as well; the init symbol (only emitted into the `cc` object, by DMD 2.076 too) isn't required as the struct is zero-initialized. When using a non-zero initializer for a struct field, both DMD and LDC fail to link due to the missing init symbol.
